algorithms/kernel/covariance/covariance_helper.h

```diff
@@ -0,0 +1,64 @@
+/* file: covariance_helper.h */
+/*******************************************************************************
+* Copyright 2014-2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef __COVARIANCE_HELPER_H__
+#define __COVARIANCE_HELPER_H__
+
+#include "service/kernel/service_environment.h"
+
+namespace daal
+{
+namespace algorithms
+{
+namespace covariance
+{
+namespace internal
+{
+template <typename algorithmFPType, CpuType cpu>
+struct BSHelper
+{
+    static size_t getBlockSize(const size_t n, const size_t k)
+    {
+        const double cacheFullness     = 0.8;
+        const size_t maxRowsPerBlock   = 512;
+        const size_t minRowsPerBlockL1 = 256;
+        const size_t minRowsPerBlockL2 = 8;
+        size_t rowsFitL1               = (services::internal::getL1CacheSize() / sizeof(algorithmFPType) * cacheFullness - n * n) / n;
+        size_t rowsFitL2               = (services::internal::getL2CacheSize() / sizeof(algorithmFPType) * cacheFullness - n * n) / n;
+        size_t blockSize               = 96;
```

Comment:
Why 96?

Reply:
That is "default" approach for doing that optimization that is used in kmeans. We will calculate optimal value for this parameter in future.

Reply:
I don't understand how k-means are related to covariance.

Reply:
This value is obtained experimentally as most favorable for cases that exceed L2$ capacity.

Reply:
The reader does not know about this. but he must understand where such magic numbers come from.

Reply:
> This value is obtained experimentally as most favorable for cases that exceed L2$ capacity.

Moreover, unless the gemm has a different form, it will not affect your experiments?

Reply:
We will check it ofc.

Reply:
I want you to add comments to these numbers later. To make it clear to anyone.

Reply:
Thanks, we will append some clarifying comments.